src/events/app/index.ts

```diff
@@ -0,0 +1,144 @@
+import { Context, Probot, ProbotOctokit } from "probot";
+import { IPullService } from "../../services/PullService";
+import { sleep } from "../../utils/util";
+
+interface RepoConfig {
+  owner: string;
+  repo: string;
+}
+
+// Triggered when the program starts.
+export async function handleAppStartUpEvent(
+  app: Probot,
+  gc: InstanceType<typeof ProbotOctokit>,
+  pullService: IPullService
+) {
+  let repoConfigs: RepoConfig[];
+
+  if (process.env.SYNC_REPOS !== undefined) {
+    repoConfigs = await getSyncRepositoryListFromEnv();
+  } else {
+    repoConfigs = await getSyncRepositoryListFromInstallation(app);
+  }
+
+  for (let repoConfig of repoConfigs) {
+    await handleSyncRepo(repoConfig, gc, pullService);
+  }
+}
+
+// Triggered when the user first installs the bot to the account.
+export function handleAppInstallOnAccountEvent(
+  context: Context,
+  pullService: IPullService
+) {
+  const { installation, repositories } = context.payload;
+  const repoConfigs = repositories.map((repository: { name: string }) => {
+    return {
+      owner: installation.account.login,
+      repo: repository.name,
+    };
+  });
+
+  for (let repoConfig of repoConfigs) {
+    handleSyncRepo(repoConfig, context.octokit, pullService).then(null);
+  }
+}
+
+// Triggered when the user installs the bot to another new repository
+// of the account, which has already installed the bot.
+export function handleAppInstallOnRepoEvent(
+  context: Context,
+  pullService: IPullService
+) {
+  const { installation, repositories_added } = context.payload;
+  const repoConfigs = repositories_added.map((repository: { name: string }) => {
+    return {
+      owner: installation.account.login,
+      repo: repository.name,
+    };
+  });
+
+  for (let repoConfig of repoConfigs) {
+    handleSyncRepo(repoConfig, context.octokit, pullService).then(null);
+  }
+}
+
+// General handling of a repo.
+async function handleSyncRepo(
+  repoConfig: RepoConfig,
+  gc: InstanceType<typeof ProbotOctokit>,
```

Comment:
Maybe github or githubClient better. Because TS has no requirement or habit of using abbreviations.

Reply:
ok